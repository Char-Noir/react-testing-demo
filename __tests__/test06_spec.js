// index06.js test it with jest, with snapshots.
import React from 'react';
import Link from '../src/index06';
import renderer from 'react-test-renderer';

test('06. Link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://alibaba-inc.com">Alibab</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});