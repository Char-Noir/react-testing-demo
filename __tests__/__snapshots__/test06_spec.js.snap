// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`06. Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://alibaba-inc.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Alibab
</a>
`;

exports[`06. Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://alibaba-inc.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Alibab
</a>
`;

exports[`06. Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://alibaba-inc.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Alibab
</a>
`;
